agent/agent.go

```diff
@@ -427,34 +428,59 @@ func (a *Agent) handlePolicy(p *policystorage.Policy) {
 		// plugins doing this.
 		action.Canonicalize()
 
-		if !action.IsWithinLimits(p.Min, p.Max) {
+		// If the policy is configured with dry-run:true then we set the
+		// action count to nil so its no-nop. This allows us to still
+		// submit the job, but not alter its state.
+		if val, ok := p.Target.Config["dry-run"]; ok && val == "true" {
```

Comment:
is the .Meta API exposed in the autoscaler yet? this would be a good candidate for some additional metadata.